### PR TITLE
Edit application from commit in App detail tab

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes https://github.com/epinio/ui/issues/197

### CODE REVIEW

File: pkg/epinio/detail/applications.vue [hunk 1]
```diff
@@ -149,7 +149,16 @@ export default Vue.extend<Data, any, any, any>({
 
       const arr: any[] = isArray(commits) ? commits : [commits];
 
-      return arr.map(c => GitUtils[this.gitType].normalize.commit(c));
+      return arr.map(c => ({
+        ...GitUtils[this.gitType].normalize.commit(c),
+        availableActions: [{
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:154]: Can this be a const assigned in `data`?

File: pkg/epinio/detail/applications.vue [hunk 2]
```diff
@@ -149,7 +149,16 @@ export default Vue.extend<Data, any, any, any>({
...
+          action:  'editFromCommit',
+          label:   this.t('epinio.applications.actions.editFromCommit.label'),
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:156]: We might need a better label for this, something to denote the action won't kick anything off straight away.

Something like `Deploy commit`  (subtle difference from the one word actions we have elsewhere). Not sure on that, open to improvements

**Review by richard-cox — CHANGES_REQUESTED** [review]
This is a bit of a pain, but when we navigate to the edit app page with the commit pre-selected we should show the page it's on. Normally this is the first page, but can be page 99 out of 100. I've created a branch in gitlab richard-cox/epinio-sample-app called lots-of-commits that can be used for testing

File: pkg/epinio/detail/applications.vue (hunk 1 above)
Comment (torchiaf) [reply pkg/epinio/detail/applications.vue:154]: Do you mean: 

```
data() {
  return {
    ...
    availableActions: [{
      action:  'editFromCommit',
      label:   this.t('epinio.applications.actions.editFromCommit.label'),
      icon:    'icon icon-edit',
      enabled: true,
    }],
  };
```

**Pull request merged**